packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -205,24 +197,37 @@ export class pTokensEvmProvider implements pTokensAssetProvider {
         return false
       }
     }, _pollingTime)
+    if (!receipt) throw new Error('Polling function stopped unexpectedly. No receipt found')
     return receipt.transactionHash.toString()
   }
 
+  /**
+   * Wait for tx receipt.
+   * @param _txHash - Transcation hash of the searched receipt.
+   * @returns TransactionReceipt of _txHash.
+   */
+  async waitForTransactionReceipt(_txHash: `0x${string}`): Promise<TransactionReceipt> {
```

Comment:
Isn't this function the the same as waitForTransactionConfirmation ?
Without breaking interface you might change waitForTransactionConfirmation implementation calling waitForTransactionReceipt inside. You maintain the same interface without creating new functions.